server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala

```diff
@@ -270,6 +270,191 @@ trait EmailQueryMethodContract {
     }
   }
 
+  @Test
+  def listMailsShouldBeSortedByDescendingOrderOfSentAt(server: GuiceJamesServer): Unit = {
+    val message: Message = Message.Builder
+      .of
+      .setSubject("test")
+      .setBody("testmail", StandardCharsets.UTF_8)
+      .build
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(MailboxPath.inbox(BOB))
+    val otherMailboxPath = MailboxPath.forUser(BOB, "other")
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(otherMailboxPath)
+    val requestDateMessage1 = Date.from(ZonedDateTime.now().minusDays(1).toInstant)
+    val messageId1: MessageId = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB),
+        AppendCommand.builder().withInternalDate(requestDateMessage1).build(message))
```

Comment:
sendDate != internalDate

You should set the date on the Mime message:

```
    val message: Message = Message.Builder
      .of
      .setSubject("test")
      .setDate(...)
      .setBody("testmail", StandardCharsets.UTF_8)
      .build
```

Reply:
We should however have a test showing that when a message have no `Date` header, we default to the internal date